build/source/engine/computFlux.f90

```diff
@@ -0,0 +1,851 @@
+ mLayerdPsi_dTheta            => deriv_data%var(iLookDERIV%mLayerdPsi_dTheta           )%dat     ,&  ! intent(out): [dp(:)] derivative in the soil water characteristic w.r.t. theta
+ dCompress_dPsi               => deriv_data%var(iLookDERIV%dCompress_dPsi              )%dat     ,&  ! intent(out): [dp(:)] derivative in compressibility w.r.t matric head
+
+ ! derivative in liquid water fluxes for the soil domain w.r.t energy state variables
+ dq_dNrgStateAbove            => deriv_data%var(iLookDERIV%dq_dNrgStateAbove           )%dat     ,&  ! intent(out): [dp(:)] change in flux at layer interfaces w.r.t. states in the layer above
+ dq_dNrgStateBelow            => deriv_data%var(iLookDERIV%dq_dNrgStateBelow           )%dat     ,&  ! intent(out): [dp(:)] change in flux at layer interfaces w.r.t. states in the layer below
+ mLayerdTheta_dTk             => deriv_data%var(iLookDERIV%mLayerdTheta_dTk            )%dat     ,&  ! intent(out): [dp(:)] derivative of volumetric liquid water content w.r.t. temperature
+ dPsiLiq_dTemp                => deriv_data%var(iLookDERIV%dPsiLiq_dTemp               )%dat      &  ! intent(out): [dp(:)] derivative in the liquid water matric potential w.r.t. temperature
+ )  ! association to data in structures
+
+ ! *****
+ ! * PRELIMINARIES... 
+ ! ******************
+
+ ! modify the groundwater representation for this single-column implementation
+ select case(ixSpatialGroundwater)
+  case(singleBasin); local_ixGroundwater = noExplicit    ! force no explicit representation of groundwater at the local scale
+  case(localColumn); local_ixGroundwater = ixGroundwater ! go with the specified decision
+  case default; err=20; message=trim(message)//'unable to identify spatial representation of groundwater'; return
+ end select ! (modify the groundwater representation for this single-column implementation)
+
+ ! check that canopy temperature is reasonable
+ if(scalarCanopyTempTrial > canopyTempMax)then
+  print*, 'scalarCanopyTempTrial = ', scalarCanopyTempTrial
+  message=trim(message)//'canopy temperature is > expected maximum'
+  err=20; return
+ endif
+
+ ! *****
+ ! * COMPUTE DERIVATIVES ASSOCIATED WITH MELT-FREEZE...
+ ! ****************************************************
+
+ ! * vegetation domain: compute derivative of volumetric liquid water content w.r.t. temperature (K-1)
+ if(computeVegFlux)then
+  if(scalarCanopyIceTrial > verySmall)then
+   theta = (scalarCanopyIceTrial + scalarCanopyLiqTrial)/(canopyDepth*iden_water)
+   dTheta_dTkCanopy = dFracLiq_dTk(scalarCanopyTempTrial,snowfrz_scale)*theta   ! K-1
+   dCanLiq_dTcanopy = dTheta_dTkCanopy*iden_water*canopyDepth                   ! kg m-2 K-1
+  else
+   dTheta_dTkCanopy = 0._dp
+   dCanLiq_dTcanopy = 0._dp
+  endif
+ endif
+
+ ! * snow+soil domain: compute derivative of volumetric liquid water content w.r.t. temperature (K-1)
+ do iLayer=1,nLayers  ! loop through all snow and soil layers
```

Comment:
Good point. There is likely substantial speedup possible through identifying all such instances. It would be good if these are all addressed together and systematically. I've opened up a new issue (issue #142) to address this directly.